src/main/java/org/cbio/gdcpipeline/writer/ClinicalWriter.java

```diff
@@ -0,0 +1,152 @@
+                if (data instanceof Patient) {
+                    filterList.add(data);
+                }
+            }
+        } else if (writerType.equals("sample")) {
+            for (ClinicalDataModel data : list) {
+                if (data instanceof Sample) {
+                    filterList.add(data);
+                }
+            }
+        }
+        clinicalWriter.write(filterList);
+    }
+
+
+    private void configureWriter(String writerType) {
+        ClinicalDataModel data = null;
+        String outputFile = outputDir + File.separator;
+        if (writerType.equals("patient")) {
+            data = new Patient();
+            outputFile += patientFile;
+        } else {
+            data = new Sample();
+            outputFile += sampleFile;
+        }
+
+        clinicalWriter.setShouldDeleteIfExists(true);
+        clinicalWriter.setLineSeparator(System.lineSeparator());
+        clinicalWriter.setHeaderCallback(clinicalDataHeader(data));
+        DelimitedLineAggregator<ClinicalDataModel> lineAggregator = new DelimitedLineAggregator<>();
+        lineAggregator.setDelimiter("\t");
+        FieldExtractor<ClinicalDataModel> fe = createFieldExtractor(data);
+        lineAggregator.setFieldExtractor(fe);
+        clinicalWriter.setLineAggregator(lineAggregator);
+        clinicalWriter.setResource(new FileSystemResource(new File(outputFile)));
+        clinicalWriter.open(this.executionContext);
+
+
+    }
+
+    private FlatFileHeaderCallback clinicalDataHeader(ClinicalDataModel data) {
+        return new FlatFileHeaderCallback() {
+            @Override
+            public void writeHeader(Writer writer) throws IOException {
+                Map<String, List<String>> headers = data.getHeaders();
+                StringBuilder sb = new StringBuilder();
+                sb.append(StringUtils.join(headers.get("displayNames"), '\t')).append("\n");
+                sb.append(StringUtils.join(headers.get("description"), '\t')).append("\n");
+                sb.append(StringUtils.join(headers.get("datatype"), '\t')).append("\n");
+                sb.append(StringUtils.join(headers.get("priority"), '\t')).append("\n");
+                List<String> list = headers.get("headers");
+                for (int i = 0; i < list.size(); i++) {
+                    list.set(i, list.get(i).toUpperCase());
+                }
+                sb.append(StringUtils.join(list, '\t'));
+                writer.write(sb.toString());
+
+            }
+        };
+    }
+
+
+    private FieldExtractor<ClinicalDataModel> createFieldExtractor(ClinicalDataModel data) {
+        BeanWrapperFieldExtractor<ClinicalDataModel> ext = new BeanWrapperFieldExtractor<>();
+        List<String> fieldList = data.getFields();
+        String[] fields = new String[fieldList.size()];
+        fields = data.getFields().toArray(fields);
+        for (int i = 0; i < fields.length; i++) {
+            fields[i] = fields[i].toLowerCase();
```

Comment:
You could set the fields to all lower case using a stream before converting to array:
`List<String> fieldList = fieldList.stream().map(String::toLowerCase).collect(Collectors.toList());`

I think the collections streams stuff is new to java 8.